hub/core/compression.py

```diff
@@ -805,311 +760,137 @@ def _read_audio_shape(
     return (info.num_frames, info.nchannels)
 
 
-def _decompress_video_cffi(
-    file,
-    compression,
-    start_frame: Optional[int] = None,
-    end_frame: Optional[int] = None,
-    step: Optional[int] = None,
-    reverse: bool = False,
-):
-    # int decompressVideo(unsigned char *file, int size, int ioBufferSize, unsigned char *decompressed, int isBytes, int nbytes)
-    # isBytes should be set to 1 in case of in-memory video else set to 0
-    # if isBytes is 1, size of file and internal buffer size must be set
-    # buffer size currently set to size of compressed data
-
-    ffmpeg_binary()  # raise error if ffmpeg not installed
-
-    from hub.core.pyffmpeg._pyffmpeg import lib, ffi  # type: ignore
-
-    if step is None:
-        step = 1
-
-    shape = _read_video_shape_cffi(file, compression)
-    n_frames = shape[0]
-    start_frame, end_frame = _norm_video_frame_indices(
-        start_frame, end_frame, reverse, n_frames
+def _frame_to_stamp(nframe, stream):
+    fps = stream.guessed_rate.numerator / stream.guessed_rate.denominator
+    seek_target = nframe / fps
+    stamp = math.floor(
+        seek_target * (stream.time_base.denominator / stream.time_base.numerator)
     )
-    n_frames = end_frame - start_frame
-
-    step_seeking = False
+    return stamp
 
-    shape = (n_frames, *shape[1:])
-    nbytes = np.prod(shape)
-    if (
-        step > 1 and nbytes > 1e8
-    ):  # if bytes to allocate video[start:end] > 100MB, use seeking
-        step_seeking = True
-        n_frames = math.ceil(n_frames / step)
-        shape = (n_frames, *shape[1:])
-        nbytes = np.prod(shape)
-
-    if n_frames <= 0:
-        return np.zeros((0,) + shape[1:], dtype=np.uint8)
-
-    decompressed = ffi.new(f"unsigned char[{nbytes}]")
 
+def _open_video(file: Union[str, bytes, memoryview]):
     if isinstance(file, str):
-        lib.decompressVideo(
-            file.encode("utf-8"),
-            0,
-            0,
-            start_frame,
-            step if step_seeking else 1,
-            decompressed,
-            0,
-            nbytes,
+        container = av.open(
+            file, options={"protocol_whitelist": "file,http,https,tcp,tls,subfile"}
         )
     else:
-        lib.decompressVideo(
-            bytes(file),
-            len(file),
-            len(file),
-            start_frame,
-            step if step_seeking else 1,
-            decompressed,
-            1,
-            nbytes,
-        )
+        container = av.open(BytesIO(file))
 
-    video = np.frombuffer(ffi.buffer(decompressed), dtype=np.uint8).reshape(shape)
+    vstreams = container.streams.video
 
-    if not step_seeking and step > 1:
-        video = video[::step].copy()
-    if reverse:
-        video = video[::-1]
-    return video
+    if len(vstreams) == 0:
+        raise IndexError("No video streams available!")
```

Comment:
when would this happen? can we test it?

Reply:
i will try to find a file that does this.